From dc46a216798ad5493f65cf02fb2e5225547993f0 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Wed, 6 Mar 2019 11:02:52 +0800
Subject: [PATCH 496/706] fsl-mc: remove the redefinition for fsl_mc_command

In file included from drivers/bus/fsl-mc/fsl-mc-private.h:11,
                 from drivers/bus/fsl-mc/fsl-mc-bus.c:22:
./include/linux/fsl/mc.h:219:8: error: redefinition of 'struct fsl_mc_command'
 struct fsl_mc_command {
        ^~~~~~~~~~~~~~
In file included from ./include/linux/fsl/mc.h:16,
                 from drivers/bus/fsl-mc/fsl-mc-private.h:11,
                 from drivers/bus/fsl-mc/fsl-mc-bus.c:22:
./include/uapi/linux/fsl_mc.h:20:8: note: originally defined here
 struct fsl_mc_command {
        ^~~~~~~~~~~~~~
drivers/bus/fsl-mc/fsl-mc-bus.c:278:3: error: 'struct bus_type' has no member named 'force_dma'
  .force_dma = true,
   ^~~~~~~~~
drivers/bus/fsl-mc/fsl-mc-bus.c:278:15: warning: initialization of 'const struct attribute_group **' from 'int' makes pointer from integer without a cast [-Wint-conversion]
  .force_dma = true,
               ^~~~

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 include/uapi/linux/fsl_mc.h | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/include/uapi/linux/fsl_mc.h b/include/uapi/linux/fsl_mc.h
index e4c8dd011807..717c3fec7610 100644
--- a/include/uapi/linux/fsl_mc.h
+++ b/include/uapi/linux/fsl_mc.h
@@ -10,18 +10,6 @@
 
 #define MC_CMD_NUM_OF_PARAMS	7
 
-/**
- * struct fsl_mc_command - Management Complex (MC) command structure
- * @header: MC command header
- * @params: MC command parameters
- *
- * Used by RESTOOL_SEND_MC_COMMAND
- */
-struct fsl_mc_command {
-	__u64 header;
-	__u64 params[MC_CMD_NUM_OF_PARAMS];
-};
-
 #define RESTOOL_IOCTL_TYPE	'R'
 #define RESTOOL_IOCTL_SEQ	0xE0
 
-- 
2.17.1

